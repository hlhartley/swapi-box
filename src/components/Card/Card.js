import React from 'react';
import '../../main.scss';

const Card = () => {
    return (
        <div className='card'>
            Card
        </div>
    )
}

export default Card;