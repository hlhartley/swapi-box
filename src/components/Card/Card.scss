.card {
    background-color: #36888d;
    color: #f2f2f2;
    width: 225px;
    padding: 10px;
    text-align: center;
    margin: 22px;
    border-radius: 15px;
}

.card-title {
    display: flex;
    justify-content: space-around;
    height: 65px;
}

.card-body {
    padding: 10px;
    background-color: #f2f2f2;
    color: black;
    margin: 10px;
}

button {
    background-color: #36888d;
    border: none;
}

.favorite-card-style {
    background-color: blue;
}