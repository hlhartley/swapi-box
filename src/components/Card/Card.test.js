import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
import { shallow } from 'enzyme';

let wrapper;
describe('Card', () => {

beforeEach(() => {
    wrapper = shallow(<Card/>); 
})

    it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Card />, div);
    ReactDOM.unmountComponentAtNode(div);
    })

    it('should exist', () => {   
    expect(wrapper).toBeDefined()
    })

    it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
    })
})

