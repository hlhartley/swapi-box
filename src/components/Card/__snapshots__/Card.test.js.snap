// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot with all data passed in correctly 1`] = `
<div
  className="card"
>
  <div
    className="card-div"
  >
    <h1
      className="card-title"
    >
      Alderaan
      <button
        onClick={[Function]}
      >
        <i
          className="far fa-star"
        />
      </button>
    </h1>
    <i
      className="fas fa-globe"
      style={
        Object {
          "fontSize": "45px",
        }
      }
    />
    <br />
    <p
      className="card-body"
    >
      Terrain: 
      grasslands, mountains
      <br />
      Population: 
      2000000000
      <br />
      Climate: 
      temperate
      <br />
      Residents: 
       https://swapi.co/api/people/5/,
       https://swapi.co/api/people/68/,
       https://swapi.co/api/people/81/,
    </p>
  </div>
</div>
`;
